IOPool/Input/src/RootPrimaryFileSequence.cc

```diff
@@ -80,7 +80,7 @@ namespace edm {
       }
     } else {
       if (!nextFile()) {
-        assert(0);
+        return std::unique_ptr<FileBlock>();
```

Comment:
Could you include also the corresponding comment from #32373?
```suggestion
        // handle case with last file bad and
        // skipBadFiles true
        return std::unique_ptr<FileBlock>();
```